lib/PuppeteerSharp/Browser.cs

```diff
@@ -210,12 +210,16 @@ public async Task<BrowserContext> CreateIncognitoBrowserContextAsync()
         /// Returns an array of all open <see cref="BrowserContext"/>. In a newly created browser, this will return a single instance of <see cref="BrowserContext"/>
         /// </summary>
         /// <returns>An array of <see cref="BrowserContext"/> objects</returns>
-        public BrowserContext[] BrowserContexts()
+        public IBrowserContext[] BrowserContexts()
         {
-            var allContexts = new BrowserContext[_contexts.Count + 1];
-            allContexts[0] = DefaultContext;
-            _contexts.Values.CopyTo(allContexts, 1);
```

Comment:
Do you have a reference issue?

Reply:
I've simplified this in commit https://github.com/hardkoded/puppeteer-sharp/pull/1965/commits/882adbacff416e724030e17c1c28300190abc9df

This should have better thread safety than the original implementation (In theory _contexts.Count could change before accessing _contexts.Values).

I'm still interested in a reference issue, I'd like to avoid any performance regressions if possible.